File: apps/mobile/app/src/main/java/org/auibar/aris/mobile/data/local/entity/CampaignEntity.kt
package org.auibar.aris.mobile.data.local.entity

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "campaigns",
    indices = [
        Index("status"),
        Index("tenantId"),
        Index("domain"),
    ],
)
data class CampaignEntity(
    @PrimaryKey val id: String,
    val tenantId: String,
    val name: String,
    val domain: String,
    val templateId: String,
    val startDate: Long,
    val endDate: Long,
    val status: String,
    val description: String? = null,
    val targetSubmissions: Int? = null,
    val totalSubmissions: Int = 0,
    val validatedSubmissions: Int = 0,
    val rejectedSubmissions: Int = 0,
    val syncedAt: Long? = null,
)

File: services/collecte/src/campaign/entities/campaign.entity.ts
export type CampaignStatusType = 'PLANNED' | 'ACTIVE' | 'COMPLETED' | 'CANCELLED';
export type ConflictStrategyType = 'LAST_WRITE_WINS' | 'MANUAL_MERGE';

export interface CampaignEntity {
  id: string;
  tenantId: string;
  name: string;
  domain: string;
  templateId: string;
  startDate: Date;
  endDate: Date;
  targetZones: string[];
  assignedAgents: string[];
  targetSubmissions: number | null;
  status: CampaignStatusType;
  description: string | null;
  conflictStrategy: ConflictStrategyType;
  dataContractId: string | null;
  createdBy: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface CampaignWithProgress extends CampaignEntity {
  progress: {
    totalSubmissions: number;
    validated: number;
    rejected: number;
    pending: number;
    completionRate: number;
  };
}
